ViAn/GUI/Bookmark/bookmarkwidget.cpp

```diff
@@ -46,10 +47,14 @@ void BookmarkWidget::generate_report() {
 
             RefDisp ref_disp = std::make_pair(_temp_disp, _temp_ref);
             rp_cont.push_back(std::make_pair(cat_name, ref_disp));
+        } else if (item->type() == BOOKMARK) {
+            BookmarkItem* bmark = dynamic_cast<BookmarkItem*>(item);
+            bmark_list.push_back(bmark);
         }
     }
     processing_thread = new QThread;    
     ReportGenerator* rp_gen = new ReportGenerator(m_path,rp_cont);
```

Comment:
Is this pointer deleted anywhere? (I know it's not part of the the new additions but if it's a dangling pointer we should fix it.)

If I remember correctly the report generation is blocking so it should be safe to remove after the `create_report` call.

Reply:
Sure it's safe to delete the rp_gen here, however that will close the word document which is not wanted. (as seen in rp_gens destructor).